lib/maths/CBoostedTree.cc

```diff
@@ -714,12 +734,14 @@ class CBoostedTree::CImpl final {
 
     std::size_t estimateMemoryUsage(std::size_t numberRows, std::size_t numberColumns) const {
         std::size_t maximumNumberNodes{this->maximumTreeSize(numberRows)};
-        std::size_t forestMemoryUsage{this->m_MaximumNumberTrees * maximumNumberNodes *
-                                      (sizeof(CNode) + sizeof(CLeafNodeStatistics))};
+        std::size_t forestMemoryUsage{this->m_MaximumNumberTrees *
+                                      maximumNumberNodes * sizeof(CNode)};
+        std::size_t leafNodeStatistics{CLeafNodeStatistics::estimateMemoryUsage(
```

Comment:
This guy needs multiplying by `maximumNumberNodes` to get an upper bound since we maintain one per leaf for the node under construction.

Reply:
done